Patches/Mechanoid Bench 3 JGH/Reverse_Patch_MB3.xml

```diff
@@ -0,0 +1,125 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<Patch>
+
+    <Operation Class="PatchOperationFindMod">
+        <mods>
+            <li>Mechanoid Bench 3</li>
+        </mods>
+        <match Class="PatchOperationSequence">
+            <operations>
+
+                <!-- Vanilla Mechanoid Weapon -->
+                <li Class="PatchOperationConditional">
+                    <xpath>/Defs/ThingDef[defName="Gun_InfernoCannon" or @Name="Gun_ChargeBlasterHeavyBase"]/destroyOnDrop</xpath>
+                    <nomatch Class="PatchOperationAdd">
+                        <xpath>/Defs/ThingDef[defName="Gun_InfernoCannon" or @Name="Gun_ChargeBlasterHeavyBase"]</xpath>
+                        <value>
+                            <destroyOnDrop>true</destroyOnDrop>
+                        </value>
+                    </nomatch>
+                </li>
+
+                <li Class="PatchOperationConditional">
+                    <xpath>/Defs/ThingDef[@Name="Gun_ChargeBlasterHeavyBase"]/tradeability</xpath>
+                    <nomatch Class="PatchOperationAdd">
+                        <xpath>/Defs/ThingDef[@Name="Gun_ChargeBlasterHeavyBase"]</xpath>
+                        <value>
+                            <tradeability>None</tradeability>
+                        </value>
+                    </nomatch>
+                    <match Class="PatchOperationReplace">
+                        <xpath>/Defs/ThingDef[@Name="Gun_ChargeBlasterHeavyBase"]/tradeability</xpath>
+                        <value>
+                            <tradeability>None</tradeability>
+                        </value>
+                    </match>
+                </li>
+
+                <li Class="PatchOperationConditional">
+                    <xpath>/Defs/ThingDef[defName="Gun_InfernoCannon"]/tradeability</xpath>
+                    <nomatch Class="PatchOperationAdd">
+                        <xpath>/Defs/ThingDef[defName="Gun_InfernoCannon"]</xpath>
+                        <value>
+                            <tradeability>None</tradeability>
+                        </value>
+                    </nomatch>
+                    <match Class="PatchOperationReplace">
+                        <xpath>/Defs/ThingDef[defName="Gun_InfernoCannon"]/tradeability</xpath>
+                        <value>
+                            <tradeability>None</tradeability>
+                        </value>
+                    </match>
+                </li>
+
+                <!-- Redundant RecipeDef -->

```

Comment:
I'm currently getting a start-up error related to it not being able to find these recipe defs. Maybe they're not actually redundant...?

Reply:
well those recipes defs will let players craft a weapon (specifically a mechanoid one), it wouldn't be pleasant to have Inferno cannon or charge blaster cannon and more to be craftable through the bench itself considering CE already (if I'm not mistaken) removed or replace those also making them gone the moment they got dropped, I'll see what I can do about it to suppress the error otherwise everything is fine

Reply:
Well, given that they're missing, something is looking for the recipe. At a glance, it looks like `MechanoidBench_Bench` has them, so you'd just need to remove them from there as well.

Reply:
makes sense, it's better that way than removing the recipeDefs itself. Much appreciated pointing it out